### PR TITLE
Several misc fixes

### PR DESCRIPTION
**Pull request by jhrozek**
A nicer error message, dependency trimming and avoid creating a local directory unless needed.

### CODE REVIEW

File: roles/keycloak/tasks/main.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@
     module: file
     dest: "{{ keycloak_local_download_dest }}"
     state: directory
+  when: not keycloak_archive_on_target
```
Comment (nkinder) [inline roles/keycloak/tasks/main.yml:15]: This local directory is also used as the local destination when generating a pkcs12 file when PEM files are provided.  Perhaps we can just AND together multiple conditions here so we catch that case as well.

**Review by nkinder — CHANGES_REQUESTED** [review]
There is an additional condition (pkcs12 generation) where we use the local directory, which we need to check for.  Aside from that, this looks like a good change.

File: roles/keycloak/tasks/main.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@
   vars:
     packages:
     - firewalld
-    - java-openjdk
+    - java-openjdk-headless
```
Comment (nkinder) [inline roles/keycloak/tasks/main.yml]: I had to modify the OpenJDK dependency to specifically use 1.8.0 in my CI patch, so I just made this change as a part of that PR.  We can either drop this patch, or I can just rebase my work on top of this if we choose to merge your work first.

**Comment by jhrozek** [comment]
I dropped the headless patch, but kept the other two as-as because PR #10 removes the other use of the `keycloak_local_download_dest` as it drops the block that generates the PKCS12 bundle. Perhaps it makes more sense to revisit this PR if/when PR#10 is merged, the changes here are more or less cosmetic after all.

**Comment by nkinder** [comment]
I think we should get PR#10 merged first, then we can merge this.

**Review by nkinder — APPROVED** [review]

**Pull request merged**